cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -244,6 +244,22 @@ class aggregate_reader_metadata {
     host_span<std::vector<size_type> const> row_group_indices,
     host_span<int const> column_schemas) const;
 
+  /**
+   * @brief Filters the row groups using row bounds (`skip_rows` and `num_rows`)
+   *
+   * @param rows_to_skip Number of rows to skip
+   * @param rows_to_read Number of rows to read
+   *
+   * @return A tuple of surviving row group indices, and two vectors of effective (trimmed) row
+   * counts and offsets across surviving row group indices respectively
+   */
+  [[nodiscard]] std::tuple<std::vector<std::vector<size_type>>,
+                           std::vector<std::vector<size_type>>,
+                           std::vector<std::vector<size_type>>>
+  apply_row_bounds_filter(cudf::host_span<std::vector<size_type> const> input_row_group_indices,
```

Comment:
Why is this function declared in the header?

Reply:
Oh it's a member of `aggregate_reader_metadata` class